src/PlatformServices/Services/TestSource.cs

```diff
@@ -21,23 +22,30 @@ public class TestSource : ITestSource
     {
         private const string SystemAssembliesPrefix = "system.";
 
-        private static IEnumerable<string> executableExtensions = new HashSet<string>()
+        /// <summary>
+        /// Gets the set of valid extensions for sources targeting this platform.
+        /// </summary>
+        public IEnumerable<string> ValidSourceExtensions
         {
-             Constants.ExeExtension,
-
-             // Required only for store 8.1. In future if that support is needed, uncomment this.
-             // Constants.DllExtension
-        };
+            get
+            {
+                return new[] {
```

Comment:
Do you really want to return a new array at every call? If not it should be `{ get;} = new[] {...}`.

Reply:
Nope - preserved the original code, will refactor this.